File: models/address.ts
import mongoose, { Schema, Document } from "mongoose";

export interface IAddress extends Document {
  streetAddress?: string;
  city?: string;
  state?: string;
  zipCode?: number;
  user?: Schema.Types.ObjectId,
}

const addressSchema = new Schema<IAddress>(
  {
    user: {
      type: Schema.Types.ObjectId,
      ref: 'users',
      cast: 'Invalid user type'
    },
    streetAddress: {
      type: String,
      cast: "address line datatype is incorrect",
    },
    city: {
      type: String,
      cast: "city datatype is incorrect",
    },
    state: {
      type: String,
      cast: "state datatype is incorrect",
    },
    zipCode: {
      type: Number,
      cast: "zipCode datatype is incorrect",
    },
  },
  { timestamps: false, versionKey: false }
);

export const addressModel = mongoose.model('addresses',addressSchema);

File: mobile_app/lib/models/address.dart
class Address {
  String? id;
  String? streetAddress;
  String? city;
  String? state;
  String? country;
  int? zipCode;

  Address({
    this.city,
    this.id,
    this.country,
    this.state,
    this.streetAddress,
    this.zipCode,
  });

  factory Address.fromJson(Map<String, dynamic> json) {
    return Address(
      id: json['_id'],
      country: json['country'],
      streetAddress: json['streetAddress'],
      city: json['city'],
      state: json['state'],
      zipCode: json['zipCode'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      '_id': id,
      'country': country,
      'streetAddress': streetAddress,
      'city': city,
      'state': state,
      'zipCode': zipCode,
    };
  }
}
